src/transformers/models/xglm/modeling_tf_xglm.py

```diff
@@ -463,19 +463,14 @@ def _prepare_decoder_attention_mask(
     ) -> tf.Tensor:
         # create causal mask
         # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-        combined_attention_mask: tf.Tensor | None = None
-        if input_shape[-1] > 1:
-            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length)
-
-        if attention_mask is not None:
-            expand_attention_mask = _expand_mask(attention_mask, tgt_len=input_shape[-1])
-            combined_attention_mask = (
-                expand_attention_mask
-                if combined_attention_mask is None
-                else expand_attention_mask + combined_attention_mask
-            )
-
-        return combined_attention_mask
+        combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length)
+        combined_attention_mask = tf.cond(
+            input_shape[-1] > 1, lambda: combined_attention_mask, lambda: tf.ones_like(combined_attention_mask)
```

Comment:
Ah, I think you're right! Hang on, let me see what I can do.

Reply:
Investigation complete! So basically, I can't actually make this function reproduce the old behaviour when we're compiling with flexible shapes, because it's just totally forbidden in TF to have a conditional where one branch returns `None` and the other branch returns a `tf.Tensor`. 

However, when I actually followed the code through to self-attention where the attention mask is used, an attention mask of `None` is just treated as an all-ones mask (i.e. neither affects the attention logits at all). Therefore, returning all-ones instead of `None` yields the same model outputs, while obeying TF's requirements for compiling conditionals.